internal/brokers/examplebroker/broker.go

```diff
@@ -368,13 +501,21 @@ func (b *Broker) handleIsAuthorized(ctx context.Context, sessionInfo sessionInfo
 	// Take into account the cancellation.
 	switch sessionInfo.selectedMode {
 	case "password":
+		switch sessionInfo.username {
+		case "user-mfa":
```

Comment:
what happens if I don’t select password for this user? I won’t have the next MFA requested, no? Similar for the other users reset password.

Also, it means that we hardcode the name logic bindings in multiple places, I suggest that we build the list of supported MFA from the start and only append the last reset password mfa here. Then, using the user fields for can/mandate reset or multiple MFA if any MFA attempt is needed.

Basically, the caller won’t have this logic.

Reply:
Right, so you will remove the need to be in password and have some counting (this will be closer to a real broker call, then)?